packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.test.js

```diff
@@ -128,101 +134,346 @@ describe('overrides plugin', () => {
+        expect(resolver.doResolve).toHaveBeenCalledWith(
+            null,
+            createRequestContextWith({
+                path: path.join(REWRITE_DIR, REQUEST_PATH + REQUEST_EXTENSION),
+                request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+            }),
+            expect.anything(),
+            expect.anything(),
+            expect.anything()
+        )
+    })
 
-    test('projectDir is normalized with forward slashes', () => {})
+    test('overrides do not return .mock files', () => {
+        // FS_READ_HASHMAP above has both index.jsx and index.mock.jsx
+        // This test checks that index.jsx is returned by the override
+        const REQUEST_PATH = `path`
+        const testRequestContext = createRequestContextWith({
+            path: path.join('.', 'node_modules', EXTENDS_TARGET),
+            request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+        })
+
+        const {resolver, callback} = setupResolverAndCallback()
+        const overridesResolver = new OverridesResolverPlugin(options)
+        overridesResolver.handleHook(testRequestContext, {}, callback, resolver)
+
+        expect(callback).toHaveBeenCalled()
+        expect(resolver.doResolve).toHaveBeenCalledWith(
+            null,
+            createRequestContextWith({
+                path: path.join(REWRITE_DIR, REQUEST_PATH, `index.jsx`),
+                request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+            }),
+            expect.anything(),
+            expect.anything(),
+            expect.anything()
+        )
+    })
+
+    test('a nested overrides folder path/to/overrides resolves correctly', () => {
+        const REQUEST_PATH = 'exists'
+        const REQUEST_EXTENSION = '.jsx'
+        const testRequestContext = createRequestContextWith({
+            path: path.join('.', 'node_modules', EXTENDS_TARGET),
+            request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+        })
+
+        const {resolver, callback} = setupResolverAndCallback()
+
+        const nestedOverridesOptions = {
+            overridesDir: '/path/to/overrides',
+            extends: ['@salesforce/retail-react-app'],
+            projectDir: PROJECT_DIR
+        }
+
+        const overridesResolver = new OverridesResolverPlugin(nestedOverridesOptions)
+        overridesResolver.handleHook(testRequestContext, {}, callback, resolver)
+
+        const nestedRewriteDir = 'src/configs/webpack/test/path/to/overrides'
+
+        expect(callback).toHaveBeenCalled()
+        expect(resolver.doResolve).toHaveBeenCalledWith(
+            null,
+            createRequestContextWith({
+                path: path.join(nestedRewriteDir, REQUEST_PATH + REQUEST_EXTENSION),
+                request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+            }),
+            expect.anything(),
+            expect.anything(),
+            expect.anything()
+        )
+    })
+})
+
+describe('OverridePlugin.isFromExtends', () => {
```

Comment:
IMO this unit test is more valuable than every else in this file, by a large factor. :)

thank you for adding this.